drivers/bluetooth/hci/ipm_stm32wb.c

```diff
@@ -96,6 +96,37 @@ static void syscmd_status_not(SHCI_TL_CmdStatus_t status)
 	BT_DBG("status:%d", status);
 }
 
+static void tryfix_event(TL_Evt_t *tev)
+{
+/*
+ * https://github.com/zephyrproject-rtos/zephyr/issues/19509
+ * Tested on BLE stack (v1.2.0)
+ * Unresolved Resolvable Private Addresses (RPA)
+ * is reported in the peer_rpa field, and not in the peer address,
+ * as it should, when this happens the peer address is set to all FFs
+ * 0A 00 01 08 01 01 FF FF FF FF FF FF 00 00 00 00 00 00 0C AA C5 B3 3D 6B ...
+ * If such message is passed to HCI core than pairing will essentially fail.
+ * Solution: Rewrite the event with the RPA in the PEER address field
+ */
+	struct bt_hci_evt_le_meta_event *mev =
+			(struct bt_hci_evt_le_meta_event *)&tev->payload;
+
+	if (tev->evtcode == BT_HCI_EVT_LE_META_EVENT &&
+			mev->subevent == BT_HCI_EVT_LE_ENH_CONN_COMPLETE) {
+		struct bt_hci_evt_le_enh_conn_complete *evt =
+				(struct bt_hci_evt_le_enh_conn_complete *)(mev + 1);
```

Comment:
Are you sure this does the right thing? Doesn't the `(mev + 1)` in translate to `mev + 1 * sizeof(*mev)`? I.e. you need a `u8_t *` typecast somewhere there, I think, e.g. :  `(struct bt_hci_evt_le_enh_conn_complete *)((u8_t *)mev + 1)`

Reply:
Which actually begs the question, did you test this? 😃

Reply:
Yeap, i've tested it. And i was expecting here see the same: pointer incrimination by size of the struct, not simply as the integer. Without any modification to pointer calculation
`BT_WARN("%p %p", mev, evt);`
```
[00:00:09.560,900] <dbg> hci_ipm.TM_EvtReceivedCb: EVT: hcievt->evtserial.evt.evtcode: 0x3e
[00:00:09.560,900] <wrn> hci_ipm: 0x20038437 0x20038438
[00:00:09.561,100] <wrn> hci_ipm: Invalid peer addr ff:ff:ff:ff:ff:ff (random)
```
Seems like GCC is too smart.
But you right that it's making sense to cast pointer to u8_t* before addition.
Updated.

Reply:
> 
> 
> Which actually begs the question, did you test this? 😃

:) Yep, I've tested it. And there is some "pointer magic"

Reply:
sizeof(struct bt_hci_evt_le_meta_event) is 1 i think. But isn't that correct, you want to move past the struct and get the bytes after that?

Reply:
> sizeof(struct bt_hci_evt_le_meta_event) is 1 i think. 

Indeed it is, and that explains why the pointer arithmetic gave the right result.

> But isn't that correct, you want to move past the struct and get the bytes after that?

Yes, you're right, I misunderstood the intent of the code since this kind of pointer arithmetic bugs happen quite easily with the special rules C has.

Reply:
Got your point. Moving by hard-coded 1 byte offset is bad idea. Updated and added sizeof instead of hard code.